rest/user_api_test.go

```diff
@@ -1345,57 +1345,182 @@ func TestUserXattrAvoidRevisionIDGeneration(t *testing.T) {
+	//				"admin_channels":["foo", "bar1"]
+	//			}`))
+	//RequireStatus(t, putResponse, 400)
+	//putResponse = rt.SendAdminRequest("PUT", "/db/_role/role1", fmt.Sprintf(userRolePayload, ``, scope1Name, collection1Name, collectionPayload))
+	//RequireStatus(t, putResponse, 400)
+}
+
+func TestPutUserCollectionAccess(t *testing.T) {
+	base.RequireNumTestDataStores(t, 2)
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyAll)
+	testBucket := base.GetPersistentTestBucket(t)
+	defer testBucket.Close()
+
+	scopesConfig := GetCollectionsConfig(t, testBucket, 2)
+	rtConfig := &RestTesterConfig{
+		CustomTestBucket: testBucket,
+		SyncFn:           `function(doc) {channel(doc.channel); access(doc.accessUser, doc.accessChannel);}`,
+		DatabaseConfig: &DatabaseConfig{
+			DbConfig: DbConfig{
+				Scopes:           scopesConfig,
+				NumIndexReplicas: base.UintPtr(0),
+				AutoImport:       base.TestUseXattrs(),
+			},
+		},
+	}
+	dataStoreNames := GetDataStoreNamesFromScopesConfig(scopesConfig)
+	scopeName, collection1Name := dataStoreNames[0].ScopeName(), dataStoreNames[0].CollectionName()
+	collection2Name := dataStoreNames[1].CollectionName()
+
+	rt := NewRestTesterMultipleCollections(t, rtConfig, 2)
+	defer rt.Close()
+	_ = rt.Context()
+
+	collectionPayload := fmt.Sprintf(`,"%s": {
+					"admin_channels":["a"]
+				}`, collection2Name)
+	userPayload := `{
+		%s
+		"admin_channels":["foo", "bar"],
+		"collection_access": {
+			"%s": {
+				"%s": {
+					"admin_channels":[%s]
+				}
+				%s
+			}
+		}
+	}`
+
+	putResponse := rt.SendAdminRequest("PUT", "/db/_user/bob", fmt.Sprintf(userPayload, `"email":"bob@couchbase.com","password":"letmein",`,
+		scopeName, collection1Name, `"foo"`, collectionPayload))
+	RequireStatus(t, putResponse, 201)
+
+	// Upsert one collection and preserve existing
+	putResponse = rt.SendAdminRequest("PUT", "/db/_user/bob", fmt.Sprintf(userPayload, `"email":"bob@couchbase.com",`,
+		scopeName, collection1Name, `"foo", "bar"`, ""))
+	RequireStatus(t, putResponse, 200)
+	getResponse := rt.SendAdminRequest("GET", "/db/_user/bob", "")
+	RequireStatus(t, getResponse, 200)
+	assert.Contains(t, getResponse.ResponseRecorder.Body.String(), collection2Name)
+
+	// Delete collection admin channels
+	putResponse = rt.SendAdminRequest("PUT", "/db/_user/bob", fmt.Sprintf(userPayload, `"email":"bob@couchbase.com",`,
+		scopeName, collection1Name, ``, ""))
+	RequireStatus(t, putResponse, 200)
+
+	getResponse = rt.SendAdminRequest("GET", "/db/_user/bob", "")
+	RequireStatus(t, getResponse, 200)
+	//responseMap := map[string]string{}
+	//json.Unmarshal([]byte(getResponse.Body.Bytes()), &responseMap)
+	//fmt.Println(responseMap["collection_access"])
+	assert.Contains(t, getResponse.ResponseRecorder.Body.String(), `"all_channels":["!"]`)
+
+	resp := rt.SendAdminRequest("PUT", "/db/_config", fmt.Sprintf(
```

Comment:
This might be replaceable by `rt.CreateDatabase()` but I don't think you need to do it for this test.

Reply:
it's used in the last commented out scenario,
	//  Hide entries for collections that are no longer part of the database for GET /_user and /_role
should I remove it?